fast_llm/layers/language_model/head.py

```diff
@@ -147,14 +149,13 @@ def _forward_backward(
         if target is not None:
             if self._config.distillation_model is None:
                 # MTP: Shift the labels
-                target = (
-                    target[self._prediction_distance : self._prediction_distance + input_.size(0),]
-                    if kwargs[TransformerKwargs.sequence_first]
-                    else target[
-                        :,
-                        self._prediction_distance : self._prediction_distance + input_.size(1),
-                    ]
+                target_slice = slice(
+                    self._prediction_distance,
+                    self._prediction_distance
+                    + input_.size(1 - kwargs[TransformerKwargs.sequence_first])
```

Comment:
this is too clever. the original version is explicit and more readable. please use if-else here.cc

Reply:
I'll see what I can do, but this is the actual bug fix for #239, so increased complexity is expected.